pkg/controller/common/certificates/pem.go

```diff
@@ -60,10 +61,29 @@ func EncodePEMPrivateKey(privateKey rsa.PrivateKey) []byte {
 func ParsePEMPrivateKey(pemData []byte) (*rsa.PrivateKey, error) {
 	block, _ := pem.Decode(pemData)
 	if block == nil {
-		return nil, errors.New("can't decode pem block")
+		return nil, errors.New("failed to parse PEM block containing private key")
 	}
-	if block.Type != "RSA PRIVATE KEY" || len(block.Headers) != 0 {
-		return nil, errors.New("pem block is not an RSA private key")
+
+	switch {
+	case block.Type == "PRIVATE KEY":
+		return parsePKCS8PrivateKey(block.Bytes)
+	case block.Type == "RSA PRIVATE KEY" && len(block.Headers) == 0:
+		return x509.ParsePKCS1PrivateKey(block.Bytes)
+	default:
+		return nil, errors.New("expected PEM block to contain an RSA private key")
+	}
+}
+
+func parsePKCS8PrivateKey(block []byte) (*rsa.PrivateKey, error) {
+	key, err := x509.ParsePKCS8PrivateKey(block)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse private key")
 	}
-	return x509.ParsePKCS1PrivateKey(block.Bytes)
+
+	rsaKey, ok := key.(*rsa.PrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("expected an RSA private key but got %t", key)
+	}
+
+	return rsaKey, nil
```

Comment:
Am I mistaken or are we missing unit tests for this? Maybe an opportunity to add one (or two)?

Reply:
The reason I didn't add any unit tests is because these functions are just thin wrappers over standard library functions. Happy to add tests if you think differently.

Reply:
OK